docs/orleans/grains/grain-references.md

```diff
@@ -0,0 +1,214 @@
+---
+title: Grain references
+description: Learn about grain references in .NET Orleans.
+ms.date: 12/10/2023
+---
+
+# Grain references
+
+Before calling a method on grain, you first need a reference to that grain. A grain reference is a proxy object that implements the same grain interface as the corresponding grain class. It encapsulates the logical identity (type and unique key) of the target grain. Grain references are used for making calls to the target grain. Each grain reference is to a single grain (a single instance of the grain class), but one can create multiple independent references to the same grain.
+
+Since a grain reference represents the logical identity of the target grain, it is independent of the physical location of the grain, and stays valid even after a complete restart of the system. Developers can use grain references like any other .NET object. It can be passed to a method, used as a method return value, etc., and even saved to persistent storage.
+
+A grain reference can be obtained by passing the identity of a grain to the <xref:Orleans.IGrainFactory.GetGrain%60%601(System.Type,System.Guid)?displayProperty=nameWithType> method, where `T` is the grain interface and `key` is the unique key of the grain within the type.
+
+The following are examples of how to obtain a grain reference of the `IPlayerGrain` interface defined above.
+
+From within a grain class:
+
+```csharp
+Guid playerId = Guid.NewGuid(); // This would typically be read from an HTTP request parameter or elsewhere.
+IPlayerGrain player = GrainFactory.GetGrain<IPlayerGrain>(playerId);
+```
+
+From Orleans client code:
+
+```csharp
+Guid playerId = Guid.NewGuid(); // This would typically be read from an HTTP request parameter or elsewhere.
+IPlayerGrain player = client.GetGrain<IPlayerGrain>(playerId);
```

Comment:
```suggestion
// This would typically be read from an HTTP request parameter or elsewhere.
Guid playerId = Guid.NewGuid();
IPlayerGrain player = client.GetGrain<IPlayerGrain>(playerId);
```